### ISSUE TITLE
HOMAD adsystem related filters are outdated

### ISSUE THREAD

**Issue by Sauerstoffdioxid** [uBlockOrigin/uAssets#3443]
### URL(s) where the issue occurs

for instance `https://www.tele5.de/tv/kalkofes-mattscheibe/premiere-klassiker` and `https://www.watchbox.de/serien/mr-bean-16253/staffel-1/mr-bean-943000.html` (I hope nothing is geo blocked)

### Describe the issue

On Tele5, the video ads from the HOMAD adsystem (a.k.a. schneevonmorgen) are showing when watching a video. After adapting the existing HOMAD filter to load on Tele5, the ads aren't shown, but the video will pause and turn black after about 10 seconds. It seems the video will not recover from this state. (You can unpause it, but it won't become visible again.)

### Versions

- Browser/version: Firefox 62
- uBlock Origin version: 1.16.20

### Settings

- only some additional EasyLists, should not be relevant to this issue

### Notes

Note that the existing filter won't possibly work on watchbox, because the Homad code (hd-main.js) is integrated into the player's js code.
However, after researching the network requests a little I found that there is always a request made to an address like `https://s3.eu-central-1.amazonaws.com/homad-global-configs-eu-fra.schneevonmorgen.com/global_config.json?cache=7716945580` (watchbox) or `https://s3.amazonaws.com/homad-global-configs-1.schneevonmorgen.com/global_config.json` (tele5). The content of the json always seems to be 
```
{
    "enabled": true, 
    "force_disabled": false
}
``` 
So after installing an addon to change request responses and basically inverting the response to be 
```
{
    "enabled": false, 
    "force_disabled": true
}
```
it now does not show any video ads anymore, while the site is still fully usable. 

tl;dr If the filter didn't redirect `hd-main.js` but insted the `global_config.json` to an altered version, HOMAD ads could be disabled for more sites with less risk of the site breaking.

**Comment by okiehsch** [comment]
`Watchbox` has been fixed in https://github.com/uBlockOrigin/uAssets/issues/845 and the fix still works for me, using Chrome or Firefox.
I added 
`||damoh.tele5.de/*$media,redirect=noopmp3-0.1s,domain=tele5.de`
which fixes the issue at
`https://www.tele5.de/tv/kalkofes-mattscheibe/premiere-klassiker`
on my end.
`

**Comment by Sauerstoffdioxid** [comment]
I guess that works. Although I like to believe my suggested fix is cleaner, as it disables the HOMAD system altogether instead of just skipping the ads.
(Btw, I mostly referenced watchbox because I noticed HOMAD was still loading)

But yes, your filters work.

**Comment by okiehsch** [comment]
Redirecting `xmlhttprequest` does not always work correctly with Firefox, which is why I prefer redirecting the media request.
I agree with you that
````
||s3.amazonaws.com/homad-global-configs.schneevonmorgen.com/global_config.json$xmlhttprequest,redirect=nooptext,domain=tele5.de
````
is the "cleaner" fix, but the main thing is that the pre-roll ad will not play.

**Comment by Sauerstoffdioxid** [comment]
Okay, so I was going to watch some more videos on Tele5 (same link as in OP) but it seems the player will now break completely after the pre-roll segment. In Firefox the video could be played sometimes, but only very rarely. In a (very) quick test with Opera I couldn't get it to work at all.
Yesterday your filter still worked. Changing the `noopmp3-0.1s` to `noopmp4-1s` had no noticable effect for me.

**Comment by okiehsch** [comment]
I can reproduce sometimes on Tele5, watchbox still works with basically the same filter.

I tried some other videos for ecample
`https://www.tele5.de/tv/star-trek/enterprise/videos/eigenarten`
and the video works half of the time, so I think this is some site issue.
````
||s3.amazonaws.com/homad-global-configs.schneevonmorgen.com/global_config.json$xmlhttprequest,redirect=nooptext,domain=tele5.de
````
fixes the issue for Chrome users, but I will comment out the old fix for now.

**Comment by Sauerstoffdioxid** [comment]
Yes, watchbox is unaffected.
With your suggested filter the ads still load (and play) in Firefox.

**Comment by okiehsch** [comment]
I know that's why I said "fixes the issue for Chrome users" and I already said that redirecting `xmlhttprequests` does not  work  with Firefox.
Firefox does not allow to redirect xmlhttprequests to data: URI

**Comment by okiehsch** [comment]
I tried some other sites using the same HOMAD configuration and the media-redirect works without issue on all of them, for example.
````
https://video.golem.de/games/21764/gods-remastered-trailer.html
````
